suites/create.js

```diff
@@ -308,6 +308,99 @@ export function runDataIntegrityProofFormatTests({
           }
         }
       });
+      describe('Algorithms', function() {
+        describe('Add Proof', function() {
+          it('Whenever this algorithm encodes strings, ' +
+            'it MUST use UTF-8 encoding.',
+          function() {
+            for(const proof of proofs) {
+              chai.expect(proof.proofValue.isWellFormed()).to.be.true;
+            }
+          });
+          it.skip('If the algorithm produces an error, the error MUST be ' +
+          'propagated and SHOULD convey the error type.',
+          function() {});
+          it('If one or more of the proof.type, proof.verificationMethod, ' +
+          'and proof.proofPurpose values is not set, an error MUST be raised ' +
+          'and SHOULD convey an error type of PROOF_GENERATION_ERROR.',
+          function() {
+            this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20one%20or%20more%20of%20the%20proof.type%2C%20proof.verificationMethod%2C%20and%20proof.proofPurpose%20values%20is%20not%20set%2C%20an%20error%20MUST%20be%20raised%20and%20SHOULD%20convey%20an%20error%20type%20of%20PROOF_GENERATION_ERROR.';
+            for(const proof of proofs) {
+              chai.expect(proof).to.contain.keys(
+                'type', 'proofPurpose', 'verificationMethod');
+            }
+          });
+          it.skip('If options has a non-null domain item, it MUST be ' +
+          'equal to proof.domain or an error MUST be raised and SHOULD ' +
+          'convey an error type of PROOF_GENERATION_ERROR.',
+          function() {});
+          it.skip('If options has a non-null challenge item, it MUST ' +
+          'be equal to proof.challenge or an error MUST be raised ' +
+          'and SHOULD convey an error type of PROOF_GENERATION_ERROR.',
+          function() {});
+        });
+        describe('Add Proof Set/Chain', function() {
+          it.skip('Whenever this algorithm encodes strings, ' +
+            'it MUST use UTF-8 encoding.',
+          function() {
+            for(const proof of proofs) {
+              chai.expect(proof.proofValue.isWellFormed()).to.be.true;
+            }
+          });
+          it('If a proof with id equal to previousProof does not ' +
+          'exist in allProofs, an error MUST be raised and SHOULD ' +
+          'convey an error type of PROOF_GENERATION_ERROR.',
+          function() {
+            this.test.link = 'https://www.w3.org/TR/vc-data-integrity/#:~:text=If%20a%20proof%20with%20id%20equal%20to%20previousProofdoes%20not%20exist%20in%20allProofs%2C%20an%20error%20MUST%20be%20raised%20and%20SHOULD%20convey%20an%20error%20type%20of%20PROOF_GENERATION_ERROR.';
+            for(const proof of proofs) {
+              if('previousProof' in proof) {
+                if(typeof proof.previousProof === 'string') {
+                  proofs.some(
+                    otherProof => otherProof.id == proof.previousProof).
+                    should.be('True',
```

Comment:
Somewhat surprisingly the previous proof does not have to be after the proof referencing it in the array, some will work just fine here. On the other hand this should probably be proofs.some(...search for proofId).should.equal(true, `Expected previousProof ${proof.previousProof}`). As written it expects the return from some to match the string 'True' which it might or this code has just never run as no issuer I'm aware of currently issues proofChains (we really need verifier tests for this). anyways like the test so far.

Reply:
implemented your suggestion